AirCasting/SearchAndFollow/CompleteScreen/Views/CompleteScreenViewModel.swift

```diff
@@ -138,60 +100,85 @@ class CompleteScreenViewModel: ObservableObject {
         exitRoute()
     }
     
+    func confirmationButtonPressed() {
+        guard externalSessionWithStreams != nil else {
+            completeButtonEnabled = false
+            self.showAlert()
+            return
+        }
+        following()
+        saveToDb()
+    }
+    
+    private func following() {
+        completeButtonEnabled = false
+        completeButtonText = "Following..."
+    }
+    
+    private func saveToDb() {
+        guard let externalSessionWithStreams = externalSessionWithStreams else {
+            assertionFailure("Confirmation button pressed when there was no session with streams")
+            return
+        }
+        
+        service.followSession(session: externalSessionWithStreams) { [weak self] result in
+            switch result {
+            case .success:
+                Log.info("Successfully followed session: \(externalSessionWithStreams.uuid)")
+                guard let self = self else { return }
+                DispatchQueue.main.async {
+                    self.completeButtonText = self.followedText
+                }
+            case .failure(let error):
+                Log.error("Following external session failed: \(error)")
+                self?.showAlert()
+            }
+        }
+    }
+    
     private func dismissView() {
         exitRoute()
     }
     
-    private func getMeasurementsAndDisplayData(_ thresholds: ThresholdsValue) {
-        let streams = [String(session.streamID)] // In the future this will be changed for Airbeam sessions, cause we will need to get other streams ids from backend
-        downloadMeasurements(streams: streams) { [weak self] result in
+    private func getMeasurementsAndDisplayData() {
+        let streams = session.stream.map(\.id)
+        
+        service.downloadMeasurements(streamsIds: streams) { [weak self] result in
             guard let self = self else { return }
             switch result {
             case .failure(let error):
                 Log.error("Failed to download session: \(error)")
+                self.showAlert()
+            case .success(let downloadedStreamsWithMeasurements):
+                guard !downloadedStreamsWithMeasurements.isEmpty else { return }
```

Comment:
don't we wanna clear `sesionStreams` and `externalSessionsStreams` in this case??